Assets/Scripts/Grid/Token.cs

```diff
@@ -32,20 +34,76 @@ public class Token : MonoBehaviour {
 		get{return _y;}
 		set{_y = value;}
 	}
+	// Returns _canAttack
+	public bool CanAttack {
+		get{return _canAttack;}
+		set{_canAttack = value;}
+	}
+	// Returns _canMove
+	public bool CanMove {
+		get{return _canMove;}
+		set{_canMove = value;}
+	}
 #endregion
 
 
+	// Called when token added to scene
+	void Start() {
+		// Initialize vars
+		CanMove = false;
+		CanAttack = false;
+	}
+
 	// Called when the token is clicked
 	void OnMouseDown() {
 		// If a unit is on the selected token
 		if(CurrentUnit != null) {
 			// Call unit's clicked function
-			CurrentUnit.Clicked();
+			CurrentUnit.Clicked(this);
 		}
 		// Else if there is no unit
 		else{
-			Debug.Log(CurrentTerrain.Name + "(" + X + "," + Y + ")");
+			// If CanMove, then you must already have a selected unit, so move
+			// Moves to this newly selected token, passing in the previously selected token
+			if(CanMove) {
+				MoveUnit(GameController.SelectedToken);
+			}
+		}
+	}
+
+	// Move unit from input prevToken to this token
+	private void MoveUnit(Token prevToken) {
+		Unit unit = prevToken.CurrentUnit;
+		// Update current unit of 2 party tokens
+		CurrentUnit = unit;
+		prevToken.CurrentUnit = null;
+		// Move unit's position
+		unit.transform.position = gameObject.transform.position;
+		// Unselect the unit
+		unit.UnselectUnit();
```

Comment:
Shouldn't unselect unit right? Once moving here, should be able to pick attack / heal target or wait?